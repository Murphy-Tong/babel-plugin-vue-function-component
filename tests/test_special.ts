import { transform } from "./translate"
import * as path from 'path'

test("special case", function () {
    expect(transform(path.resolve(__dirname, 'cases_special', '1.tsx'), {}, true)!.code).toMatchSnapshot()
})


