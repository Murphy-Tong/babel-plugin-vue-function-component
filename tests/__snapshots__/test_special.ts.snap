// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`special case 1`] = `
"import { createTextVNode as _createTextVNode, createVNode as _createVNode, defineComponent } from "vue";
import { CreditStatus } from '@/api/krd';
import TitleBar from '@/components/title-bar';
import { KEY_REFRESH_KRD_LIST } from '@/constant';
import { openPage } from '@/utils/navigator';
import { getDefaultPool } from '@/utils/taskPool';
import Tab from '@souche-ui/vant-next/es/tab';
import Tabs from '@souche-ui/vant-next/es/tabs';
import '@souche-ui/vant-next/lib/tabs/index.less';
import KrdList from './components/list';
import styles from './index.module.less';
const TABS = [{
  name: '全部',
  type: CreditStatus.ALL
}, {
  name: '申请预授信',
  type: CreditStatus.ON_PREPARE
}, {
  name: '申请贷款',
  type: CreditStatus.ON_LOAN
}, {
  name: '已取消',
  type: CreditStatus.CANCEL
}];
export default defineComponent({
  name: "1",
  setup: function () {
    const goSelectCar = function () {
      openPage('/krd/cars');
    };
    const activeTabRef = ref('全部');
    const refreshKey = ref(1);
    onMounted(function () {
      getDefaultPool().storeById(KEY_REFRESH_KRD_LIST, function () {
        activeTabRef.value = '全部';
        refreshKey.value = refreshKey.value + 1;
      });
    });
    onUnmounted(function () {
      getDefaultPool().remove(KEY_REFRESH_KRD_LIST);
    });
    return function () {
      return _createVNode("div", {
        "class": styles.krd
      }, [_createVNode(TitleBar, null, null), _createVNode(Tabs, {
        "class": styles.tabContent,
        'active': activeTabRef.value,
        "onUpdate:active": $event => activeTabRef.value = $event,
        "key": refreshKey.value
      }, {
        default: () => [TABS.map(t => {
          return _createVNode(Tab, {
            "title": t.name,
            "key": t.name
          }, {
            default: () => [_createVNode(KrdList, {
              "class": styles.krdList,
              "type": t.type
            }, null)]
          });
        })]
      }), _createVNode("div", {
        "class": styles.bottom
      }, [_createVNode("div", {
        "class": styles.button,
        "onClick": goSelectCar
      }, [_createTextVNode("\\u65B0\\u589E\\u5BC4\\u552E\\u8F66\\u8F86")])])]);
    };
  }
});"
`;
